rpcserver.go

```diff
@@ -3058,17 +3063,24 @@ func marshalLeafKey(leafKey universe.LeafKey) *unirpc.AssetKey {
 // Taproot Asset commitment, and script_key is the script_key of the asset
 // within the Taproot Asset commitment for the given asset_id or group_key.
 func (r *rpcServer) AssetLeafKeys(ctx context.Context,
-	req *unirpc.ID) (*unirpc.AssetLeafKeyResponse, error) {
+	req *unirpc.AssetLeafKeysRequest) (*unirpc.AssetLeafKeyResponse, error) {
 
-	universeID, err := UnmarshalUniID(req)
+	universeID, err := UnmarshalUniID(req.Id)
 	if err != nil {
 		return nil, err
 	}
 
 	// TODO(roasbeef): tell above if was tring or not, then would set
 	// below diff
 
-	leafKeys, err := r.cfg.UniverseArchive.UniverseLeafKeys(ctx, universeID)
+	leafKeys, err := r.cfg.UniverseArchive.UniverseLeafKeys(
```

Comment:
I think requests with a limit over the maximum page size (which we can define in `universe/interface` or elsewhere) should be rejected here at the RPC layer. Should also assert that offset is non-negative.